core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/QualificationAutoTuner.scala

```diff
@@ -33,23 +33,50 @@ import org.apache.spark.sql.rapids.tool.qualification.{QualificationAppInfo, Qua
 class QualificationAutoTuner(val appInfoProvider: QualAppSummaryInfoProvider,
     val tunerContext: TunerContext) {
 
+  // When enabled, the profiler recommendations should only include updated settings.
+  private val filterByUpdatedPropsEnabled: Boolean = false
+
   private def writeTuningReport(tuningResult: TuningResult,
       outputDir: String, hadoopConf: Configuration): Unit = {
+    // First, write down the recommendations and the comments
     val textFileWriter = new ToolTextFileWriter(outputDir,
-      s"${tuningResult.appID}.log", s"Tuning Qual app - ${tuningResult.appID}", Option(hadoopConf))
+      s"${tuningResult.appID}.log", s"Tuning Qual App - ${tuningResult.appID}", Option(hadoopConf))
     try {
-      textFileWriter.write(s"### Recommended Configuration for App: ${tuningResult.appID} ###\n")
+      textFileWriter.write(
+        s"### Recommended SPARK Configuration on GPU Cluster for App: ${tuningResult.appID} ###\n")
       textFileWriter.write(Profiler.getAutoTunerResultsAsString(
         tuningResult.recommendations, tuningResult.comments))
     } finally {
       textFileWriter.close()
     }
+    // Write down the combined configurations
+    tuningResult.combinedProps.collect {
+      case combinedProps =>
+        val textFileWriter = new ToolTextFileWriter(outputDir,
+          s"${tuningResult.appID}.conf",
+          s"Qual combined configurations for App - ${tuningResult.appID}", Option(hadoopConf))
+        try {
+          textFileWriter.write(combinedProps.map(_.toString).reduce(_ + "\n" + _))
+        } finally {
+          textFileWriter.close()
+        }
+    }
   }
+
   def runAutoTuner(): TuningResult = {
     val autoTuner: AutoTuner = AutoTuner.buildAutoTuner(
       tunerContext.workerInfoPath, appInfoProvider, tunerContext.platform)
-    val (recommendations, comments) = autoTuner.getRecommendedProperties()
-    val resultRecord = TuningResult(appInfoProvider.getAppID, recommendations, comments)
+    val (recommendations, comments) =
+      autoTuner.getRecommendedProperties(showOnlyUpdatedProps = filterByUpdatedPropsEnabled)
+    // Combine the GPU recommendations with all others.
+    // There are two ways we can that:
+    // 1- Combine them from the beginning; Or
+    // 2- At the end, get the union of the two properties.
```

Comment:
I prefer the second method as well. It's easier to distinguish between Spark properties and GPU recommendations.

Reply:
+1